src/Analyser/ConstantResolver.php

```diff
@@ -116,25 +121,25 @@ public function resolvePredefinedConstant(string $resolvedConstantName): ?Type
 			$maxRelease = null;
 
 			if (
-				$this->composerMinPhpVersion !== null
-				&& $this->composerMaxPhpVersion !== null
-				&& $this->composerMaxPhpVersion->getMajorVersionId() === $this->composerMinPhpVersion->getMajorVersionId()
-				&& $this->composerMaxPhpVersion->getMinorVersionId() === $this->composerMinPhpVersion->getMinorVersionId()
+				$this->getMinPhpVersion() !== null
+				&& $this->getMaxPhpVersion() !== null
+				&& $this->getMaxPhpVersion()->getMajorVersionId() === $this->getMinPhpVersion()->getMajorVersionId()
+				&& $this->getMaxPhpVersion()->getMinorVersionId() === $this->getMinPhpVersion()->getMinorVersionId()
 			) {
-				$minRelease = $this->composerMinPhpVersion->getPatchVersionId();
-				$maxRelease = $this->composerMaxPhpVersion->getPatchVersionId();
+				$minRelease = $this->getMinPhpVersion()->getPatchVersionId();
```

Comment:
You're calling the getter multiple times and it always leads to creating new object. I'd just do `if (in_array($resolvedConstantName, ['PHP_VERSION_ID', 'PHP_MAJOR_VERSION', 'PHP_MINOR_VERSION']))` and save it to a variable first, then use the variable. inside for each constant.